### DISCUSSION TITLE
Same user in multiple identity providers

### DISCUSSION THREAD

**Discussion by japoorv** [keycloak/keycloak#22783] [Q&A]
We want to provide our users the ability to login via different identity providers (a single user should be able to login by say google, facebook both). 
We maintain the user data object in our db with email as the identifier, so as long as the email is same we know that the user is same even if he logged in via two different IdP. 

However if the user has first logged in via Google, and then logs in via Facebook it asks us to link the account. Is it possible that we can create a different account for this user (with the old account having Google as social provider, newer one having facebook as social provider).

**Comment by justin-tay** [comment]
It is possible, but I'm not sure why you would want to do this.

This is related to the `Create User If Unique` step in the `first broker login` flow implemented in the `IdpCreateUserIfUniqueAuthenticator`.

You can go to `Realm settings > Login` and set the `Email settings` to
- `Email as username`: Off
- `Login with email`: Off
- `Duplicate emails`: On

If that's not sufficient you'd need to write your own authenticator.

https://github.com/keycloak/keycloak/blob/82a94cb12b3fdeff8f3c8bcddaad09dcb05ce1a2/services/src/main/java/org/keycloak/authentication/authenticators/broker/IdpCreateUserIfUniqueAuthenticator.java#L120-L135
